### PR TITLE
Allows for custom email templates

### PR DESCRIPTION
**Pull request by whitdog47**
This PR allows for custom email templates starting with the welcome email. Users select the email template, welcome message, and components. If a custom email template does not exist or is not enabled, the default template will be used.

image

### CODE REVIEW

File: src/dispatch/email_templates/service.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    return (
+        db_session.query(EmailTemplates)
+        .filter(EmailTemplates.project_id == project_id)
+        .filter(EmailTemplates.email_template_type == email_template_type)
+        .filter(EmailTemplates.enabled == True)  # noqa
+        .first()
+    )
+
+
+def get_all(*, db_session: Session):
```
Comment (metroid-samus) [inline src/dispatch/email_templates/service.py:32]: Nit
```suggestion
def get_all(*, db_session: Session) -> List[Optional[EmailTemplates]]:
```

File: src/dispatch/messaging/strings.py [hunk 2]
```diff
@@ -915,3 +916,42 @@ def render_message_template(message_template: List[dict], **kwargs):
         data.append(d)
 
     return data
+
+
+def generate_welcome_message(welcome_message: EmailTemplates):
```
Comment (metroid-samus) [inline src/dispatch/messaging/strings.py:921]: Nit
```suggestion
def generate_welcome_message(welcome_message: EmailTemplates): -> List[dict]:
```

File: src/dispatch/incident/messaging.py [hunk 3]
```diff
@@ -205,11 +219,12 @@ def send_welcome_email_to_participant(
     notification_text = "Incident Notification"
 
     # Can raise exception "tenacity.RetryError: RetryError". (Email may still go through).
+    print(f"**** Trying to send message and welcome_template is {welcome_template}")
```
Comment (metroid-samus) [inline src/dispatch/incident/messaging.py:222]:
```suggestion
    log.debug(f"**** Trying to send message and welcome_template is {welcome_template}")
```

File: src/dispatch/incident/messaging.py [hunk 4]
```diff
@@ -270,14 +283,35 @@ def send_completed_form_email(
...
 ):
     """Sends welcome messages to the participant."""
+    # check to see if there is an override welcome message template
+    welcome_template = email_template_service.get_by_type(
+        db_session=db_session,
+        project_id=incident.project_id,
+        email_template_type=EmailTemplateTypes.welcome,
+    )
+    if welcome_template is None:
+        print(f"**** No welcome_template found for {incident.project_id} with type {EmailTemplateTypes.welcome}")
```
Comment (metroid-samus) [inline src/dispatch/incident/messaging.py:298]:
```suggestion
        log.debug(f"**** No welcome_template found for {incident.project_id} with type {EmailTemplateTypes.welcome}")
```

File: src/dispatch/incident/messaging.py (hunk 3 above)
Comment (whitdog47) [reply src/dispatch/incident/messaging.py:222]: oops - leftover logging - will remove.

File: src/dispatch/incident/messaging.py (hunk 4 above)
Comment (whitdog47) [reply src/dispatch/incident/messaging.py:298]: oops - leftover logging - will remove.

**Review by metroid-samus — APPROVED** [review]

**Pull request merged**